projects/aca-content/src/lib/services/modal-ai.service.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { AppConfigValues, StorageService, UnsavedChangesDialogComponent } from '@alfresco/adf-core';
+import { MatDialog } from '@angular/material/dialog';
+
+@Injectable({ providedIn: 'root' })
+export class ModalAiService {
+  private route = inject(ActivatedRoute);
+  private dialog = inject(MatDialog);
+  private storageService = inject(StorageService);
+
+  openUnsavedChangesModal(callback: () => void): void {
+    const hasPreviousSearch = this.route.snapshot?.queryParams?.query?.length > 0;
+    const modalHidden = this.storageService.getItem(AppConfigValues.UNSAVED_CHANGES_MODAL_HIDDEN) === 'true';
```

Comment:
I see one bug. Please look at below screenshot and description below that: 
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/20d68564-2ac4-4ae6-af67-ae4637d9d532">

If you check checkbox then there is added following property to storage: 
unsaved_changes__modal_hidden: true 

There is one big problem with that: please look at rest of properties in storage, some of them are prefixed by username if users may want different values for them. I believe here we have that case because not every user may want to skip this dialog. If we have for example 2 users which use the same machine then we can expect that one may want to have this dialog and second may want to skip this.

There is easy way to do this. Use "set" function from UserPreferencesService (this service can be imported from ADF's core). That services allows you to add this property to storage but additionally it prefixes it with username automatically.

Reply:
thank you for fixing that :)